### PR TITLE
feat: Deployment template (Base + Override) - Compare with version default values

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes will enable the other values options in compare with dropdown under compare values view.

Screenshot 2023-01-28 at 2 24 12 AM

Fixes - https://github.com/devtron-labs/devtron/issues/2916

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by visiting App config of different Devtron apps (existing + new) and execute different test cases.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/deploymentConfig/DeploymentTemplateView.tsx [hunk 1]
```diff
@@ -416,10 +420,15 @@ const CompareWithDropdown = ({
             })
 
             if (!selectedOption) {
```
Comment (shivani170) [inline src/components/deploymentConfig/DeploymentTemplateView.tsx:422]: In line 413 we are already using (!selectedOption) this check so can we merge these two conditions

File: src/components/deploymentConfig/DeploymentTemplateView.tsx [hunk 2]
```diff
@@ -580,31 +591,55 @@ export const DeploymentTemplateEditorView = ({
...
 
     useEffect(() => {
         if (selectedChart && selectedOption && selectedOption.id !== -1 && !fetchedValues[selectedOption.id]) {
             setFetchingValues(true)
             const isEnvOption = selectedOption.kind === 'env'
-            const _getDeploymentTemplate =
-                isEnvOverride || isEnvOption
-                    ? getEnvDeploymentTemplate(appId, isEnvOption ? selectedOption.id : envId, selectedOption.value)
-                    : getDeploymentTemplate(+appId, +selectedOption.value)
+            const isChartVersionOption = selectedOption.kind === 'chartVersion'
```
Comment (shivani170) [inline src/components/deploymentConfig/DeploymentTemplateView.tsx:620]: Can we keep all kind in constant?

File: src/components/deploymentConfig/DeploymentTemplateView.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/deploymentConfig/DeploymentTemplateView.tsx:422]: Can be merged but need to perform additional checks.

Comment (sohel-rp) [reply src/components/deploymentConfig/DeploymentTemplateView.tsx]: Done... Re-wrote in simplified lines with comments.

File: src/components/deploymentConfig/DeploymentTemplateView.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/deploymentConfig/DeploymentTemplateView.tsx]: Moved this & others to const that can be moved safely

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**